x/gamm/keeper/swap_test.go

```diff
@@ -114,13 +118,145 @@ func (suite *KeeperTestSuite) TestBalancerPoolSimpleSwapExactAmountIn() {
 				tradeAvgPrice := test.param.tokenIn.Amount.ToDec().Quo(tokenOutAmount.ToDec())
 				suite.True(tradeAvgPrice.GT(spotPriceBefore) && tradeAvgPrice.LT(spotPriceAfter), "test: %v", test.name)
 			} else {
-				_, err := keeper.SwapExactAmountIn(ctx, suite.TestAccs[0], poolId, test.param.tokenIn, test.param.tokenOutDenom, test.param.tokenOutMinAmount)
+				_, err := keeper.SwapExactAmountIn(ctx, suite.TestAccs[0], pool, test.param.tokenIn, test.param.tokenOutDenom, test.param.tokenOutMinAmount, swapFee)
 				suite.Error(err, "test: %v", test.name)
 			}
 		})
 	}
 }
 
+// TestCalcOutAmtGivenIn only tests that balancer and stableswap pools are type casted correctly while concentratedliquidity pools fail
+func (suite *KeeperTestSuite) TestCalcOutAmtGivenIn() {
+	type param struct {
+		poolType      string
+		tokenIn       sdk.Coin
+		tokenOutDenom string
+	}
+
+	tests := []struct {
+		name       string
+		param      param
+		expectPass bool
+	}{
+		{
+			name: "balancer",
+			param: param{
+				poolType:      "balancer",
+				tokenIn:       sdk.NewCoin("foo", sdk.NewInt(100000)),
+				tokenOutDenom: "bar",
+			},
+			expectPass: true,
+		},
+		{
+			name: "stableswap",
+			param: param{
+				poolType:      "stableswap",
+				tokenIn:       sdk.NewCoin("foo", sdk.NewInt(100000)),
+				tokenOutDenom: "bar",
+			},
+			expectPass: true,
+		},
+	}
+
+	for _, test := range tests {
+		suite.Run(test.name, func() {
+			// Init suite for each test.
+			suite.SetupTest()
+			keeper := suite.App.GAMMKeeper
+			ctx := suite.Ctx
+
+			var pool swaproutertypes.PoolI
+			if test.param.poolType == "balancer" {
+				poolId := suite.PrepareBalancerPool()
+				poolExt, err := suite.App.GAMMKeeper.GetPool(suite.Ctx, poolId)
+				suite.NoError(err)
+				pool = poolExt.(swaproutertypes.PoolI)
+			} else if test.param.poolType == "stableswap" {
+				poolId := suite.PrepareBasicStableswapPool()
+				poolExt, err := suite.App.GAMMKeeper.GetPool(suite.Ctx, poolId)
+				suite.NoError(err)
+				pool = poolExt.(swaproutertypes.PoolI)
+			}
+
+			swapFee := pool.GetSwapFee(suite.Ctx)
+
+			_, err := keeper.CalcOutAmtGivenIn(ctx, pool, test.param.tokenIn, test.param.tokenOutDenom, swapFee)
+
+			if test.expectPass {
+				suite.Require().NoError(err)
+			} else {
+				suite.Require().Error(err)
+			}
+		})
+	}
+}
+
+// TestCalcInAmtGivenOut only tests that balancer and stableswap pools are type casted correctly while concentratedliquidity pools fail
```

Comment:
Can we either remove the concentratedliquidity portion from the description, or add a todo to add this to this test as well as the previous one?

Reply:
added TODO, including here: https://github.com/osmosis-labs/osmosis/issues/3556